src/MessageSource.php

```diff
@@ -35,6 +35,11 @@ public function getMessage(string $id, string $category, string $locale, array $
         return dngettext($category, $id, $id, $n);
     }
 
+    public function getMessages(string $category, string $locale): array
+    {
+        throw new \RuntimeException('Function not allowed for this package.');
```

Comment:
```suggestion
        throw new \RuntimeException('Unable to get all messages from gettext.');
```

Reply:
But overall we'd better to provide a way to extract messages from gettext or make sure gettext can handle it.

Reply:
@darkdef will disucss that with @Mister-42.

Reply:
So far I indeed have been using gettext to handle extracting.

```bash
find src/ -name *.php | xargs xgettext --from-code=utf-8 --language=PHP --no-location --omit-header --sort-output --keyword=translate --output="locales/category.pot"
```

Out of laziness I just started using gettext in php, but the .pot files seems to be easy to convert to a PHP array if required.

Reply:
OK. That should be put into docs.

Reply:
Note: gettext extraction is not aware of the Yii way of plurals. I am not sure how much this impacts etraction, as I am using PHP gettext natively